#! /usr/bin/env python


PACKAGE='speed_controller'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        Name          	Type     Level Description			Default Min	Max
gen.add("update_param_",       bool_t,   0,   "update_param",       False)
gen.add("speed_kp", 	double_t,	0,  "speed_kp", 		600.0,  0,  1000.0)
gen.add("speed_ki", 	double_t,	0,  "speed_ki", 		3.0,	0,	20.0)
gen.add("speed_kd", 	double_t,	0,  "speed_kd", 		0.0,	0,	50.0)
gen.add("speed_max",	int_t,		0,	"speed_max",		20,		10,	50)

exit(gen.generate(PACKAGE, "speed_controller_test", "speed_controller_test"))
